biz.aQute.bndlib/src/aQute/bnd/osgi/resource/CapReq.java

```diff
@@ -48,14 +51,17 @@ public Resource getResource() {
 
 	@Override
 	public int hashCode() {
+		if (hashCode != 0) {
```

Comment:
Albeit unlikely, the `hashCode` could be calculated as `0`, which negates any performance optimization lazily calculating provides. Just calculate in the constructor and be done with it.